api.planx.uk/saveAndReturn/validateSession.test.ts

```diff
@@ -1,25 +1,172 @@
 import supertest from "supertest";
 import app from "../server";
+import { queryMock } from "../tests/graphqlQueryMock";
+import {
+  mockLowcalSession,
+  mockFindSession,
+  mockNotFoundSession,
+  mockGetMostRecentPublishedFlow,
+  mockGetPublishedFlowByDate,
+  stubInsertReconciliationRequests,
+  stubUpdateLowcalSessionData,
+} from "../tests/mocks/saveAndReturnMocks";
+import type { Flow } from "../types";
 
-const ENDPOINT = "/validate-session";
+const validateSessionPath = "/validate-session";
 
 describe("Validate Session endpoint", () => {
+  afterEach(() => {
+    queryMock.reset();
+  });
 
   it("throws an error if required data is missing", async () => {
-    const missingEmail = { payload: { sessionId: 123 } };
-    const missingSessionId = { payload: { email: "test" } };
+    queryMock.mockQuery(mockFindSession);
+
+    const missingEmail = {
+      payload: { sessionId: mockLowcalSession.id },
+    };
+    const missingSessionId = { payload: { email: mockLowcalSession.email } };
 
     for (const invalidBody of [missingEmail, missingSessionId]) {
       await supertest(app)
-        .post(ENDPOINT)
+        .post(validateSessionPath)
         .send(invalidBody)
         .expect(400)
-        .then(response => {
-          expect(response.body).toHaveProperty("error", "Required value missing");
-        })
+        .then((response) => {
+          expect(response.body).toHaveProperty(
+            "error",
+            "Required value missing"
+          );
+        });
     }
   });
 
-  it.todo("Returns a 200 OK for a valid session");
-  it.todo("Returns a 404 NOT FOUND for an invalid session");
-});
\ No newline at end of file
+  it("returns a 404 NOT FOUND if the session is not found", async () => {
+    queryMock.mockQuery(mockNotFoundSession);
+
+    const data = {
+      payload: {
+        sessionId: "not-found-id",
+        email: mockLowcalSession.email,
+      },
+    };
+    await supertest(app)
+      .post(validateSessionPath)
+      .send(data)
+      .expect(404)
+      .then((response) => {
+        expect(response.body).toHaveProperty(
+          "error",
+          "Unable to find your session"
+        );
+      });
+  });
+
+  it("returns a 200 OK for a valid session", async () => {
+    mockQueryWithFlowDiff({ oldFlow: {}, newFlow: {} });
+
+    const data = {
+      payload: {
+        sessionId: mockLowcalSession.id,
+        email: mockLowcalSession.email,
+      },
+    };
+
+    const expectedMessage = "No content changes since last save point";
+
+    await supertest(app)
+      .post(validateSessionPath)
+      .send(data)
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toHaveProperty("alteredNodes", null);
+        expect(response.body).toHaveProperty("message", expectedMessage);
+        expect(response.body).toHaveProperty("removedBreadcrumbs", null);
+      });
+  });
+
+  it("shows a diff for changed nodes", async () => {
```

Comment:
nit: i'd find these positive test descriptions slightly more helpful if they referenced the specific response property they're testing

eg - this one is maybe more like `it("populates alteredNodes with the diff of changed nodes",...`

and then maybe `it.todo("populates removedBreadcrumbs when alteredNodes intersects with the session breadcrumbs")`? (happy for this case to be addressed later!)